components/org.wso2.carbon.identity.organization.management.service/src/test/java/org/wso2/carbon/identity/organization/management/service/OrganizationManagerImplTest.java

```diff
@@ -743,29 +745,31 @@ public void testGetOrganizationIdToNameMap() throws OrganizationManagementExcept
         Assert.assertEquals(orgIdToNameMap.get(ORG3_ID), ORG3_NAME);
     }
 
-    @Test
-    public void testGetOrganizationIdToNameMapWithInvalidOrgIds() throws OrganizationManagementException {
-
-        TestUtils.mockCarbonContext(SUPER_ORG_ID);
-
-        List<String> orgIds = new ArrayList<>();
-        orgIds.add("Invalid_org_id");
+    @DataProvider(name = "dataForGetOrganizationIdToNameMapWithInvalidInput")
+    public Object[][] dataForGetOrganizationIdToNameMapWithInvalidInput() {
 
-        Map<String, String> orgIdToNameMap = organizationManager.getOrganizationIdToNameMap(orgIds);
-        Assert.assertNotNull(orgIdToNameMap);
-        Assert.assertEquals(orgIdToNameMap.size(), 0);
+        return new Object[][]{
+                { Collections.emptyList() },
+                { Collections.singletonList("Invalid_org_id") },
+                { Arrays.asList("Invalid_org_id_1", SUPER_ORG_ID) },
+                { Arrays.asList(SUPER_ORG_ID, "Invalid_org_id_1") }
+        };
     }
 
-    @Test
-    public void testGetOrganizationIdToNameMapWithEmptyOrgList() throws OrganizationManagementException {
+    @Test(dataProvider = "dataForGetOrganizationIdToNameMapWithInvalidInput")
+    public void testGetOrganizationIdToNameMapWithInvalidInput(List<String> orgIds) throws OrganizationManagementException {
 
         TestUtils.mockCarbonContext(SUPER_ORG_ID);
 
-        List<String> orgIds = new ArrayList<>();
-
         Map<String, String> orgIdToNameMap = organizationManager.getOrganizationIdToNameMap(orgIds);
         Assert.assertNotNull(orgIdToNameMap);
-        Assert.assertEquals(orgIdToNameMap.size(), 0);
+
+        if (orgIds.size() <= 1) {
+            Assert.assertEquals(orgIdToNameMap.size(), 0);
+        } else {
```

Comment:
do not write logics inside the test class
Pass the expected result as well to the test method through data provider and assert the output

Reply:
For the test method pass, orgIdList and expectedOrgIdToNameMap